sdk/tests/types/transaction_regular_essence.rs

```diff
@@ -470,3 +470,63 @@ fn duplicate_output_foundry() {
         Err(Error::DuplicateOutputChain(ChainId::Foundry(foundry_id_0))) if foundry_id_0 == foundry_id
     ));
 }
+
+#[test]
+fn more_than_64_same_native_tokens() {
+    let protocol_parameters = protocol_parameters();
+    let transaction_id = TransactionId::new(prefix_hex::decode(TRANSACTION_ID).unwrap());
+    let input = Input::Utxo(UtxoInput::new(transaction_id, 0).unwrap());
+    let bytes: [u8; 32] = prefix_hex::decode(ED25519_ADDRESS_1).unwrap();
+    let address = Address::from(Ed25519Address::new(bytes));
+    let amount = 1_000_000;
+    let alias_id = AliasId::from(bytes);
+    let token_scheme = TokenScheme::Simple(SimpleTokenScheme::new(70, 0, 100).unwrap());
+    let foundry_id = FoundryId::build(&AliasAddress::from(alias_id), 1, token_scheme.kind());
+    let token_id = TokenId::from(foundry_id);
+    let basic = BasicOutput::build_with_amount(amount)
+        .add_native_token(NativeToken::new(token_id, 70).unwrap())
+        .add_unlock_condition(AddressUnlockCondition::new(address))
+        .finish_output(protocol_parameters.token_supply())
+        .unwrap();
+
+    let essence = RegularTransactionEssence::builder(protocol_parameters.network_id(), rand_inputs_commitment())
+        .with_inputs([input])
+        .with_outputs(vec![basic; 100])
+        .finish_with_params(&protocol_parameters);
+
+    assert!(essence.is_ok());
+}
+
+#[test]
+fn more_than_64_distinctive_native_tokens() {
+    let protocol_parameters = protocol_parameters();
+    let transaction_id = TransactionId::new(prefix_hex::decode(TRANSACTION_ID).unwrap());
+    let input = Input::Utxo(UtxoInput::new(transaction_id, 0).unwrap());
+    let bytes: [u8; 32] = prefix_hex::decode(ED25519_ADDRESS_1).unwrap();
+    let address = Address::from(Ed25519Address::new(bytes));
+    let amount = 1_000_000;
+
+    let mut outputs = Vec::new();
+    for _ in 0..65 {
+        let alias_id = AliasId::from(rand_bytes_array());
+        let token_scheme = TokenScheme::Simple(SimpleTokenScheme::new(70, 0, 100).unwrap());
+        let foundry_id = FoundryId::build(&AliasAddress::from(alias_id), 1, token_scheme.kind());
+        let token_id = TokenId::from(foundry_id);
+        let basic = BasicOutput::build_with_amount(amount)
+            .add_native_token(NativeToken::new(token_id, 70).unwrap())
+            .add_unlock_condition(AddressUnlockCondition::new(address))
+            .finish_output(protocol_parameters.token_supply())
+            .unwrap();
+        outputs.push(basic);
+    }
+
+    let essence = RegularTransactionEssence::builder(protocol_parameters.network_id(), rand_inputs_commitment())
+        .with_inputs([input])
+        .with_outputs(outputs)
+        .finish_with_params(&protocol_parameters);
+
+    assert!(matches!(
+        essence,
+        Err(Error::InvalidTransactionNativeTokensCount(count)) if count == 65
+    ));
```

Comment:
can we remove the extra binding?

Reply:
I think that would be less clear and we also have it like this for all the tests above